cmd/eksctl-anywhere/cmd/upgrademanagementcomponents.go

```diff
@@ -85,7 +96,7 @@ var upgradeManagementComponentsCmd = &cobra.Command{
 		}
 
 		validator := management.NewUMCValidator(managementCluster, deps.Kubectl)
-		return runner.Run(ctx, clusterSpec, managementCluster, validator)
+		return runner.Run(ctx, bundles, eksaRelease, clusterSpec, managementCluster, validator)
```

Comment:
It feels strange that bundles, eksaRelease are passed here while clusterSpec contains them, can we not changing the function's parameters but modify the bundles and eksaRelease in clusterSpec?

And, did you verify the fix e2e besides unit testing?

Reply:
The goal here is to eventually remove the reliance or need to pass the new clusterSpec because building it will throw an error if the customer's cluster is using an unsupported version in the new version of the CLI.

So, I chose to pass them individually with that in mind.